framework/physics/SolverBase/chi_solver.cc

```diff
@@ -32,6 +32,11 @@ chi::InputParameters Solver::GetInputParameters()
   params.AddOptionalParameter(
     "timestep_controller", 0, "Timestep controller to use for timestepping.");
 
+  params.AddOptionalParameterBlock(
+    "time_controls",
+    chi::ParameterBlock{},
+    "Parameters to pass to the time_controller.");
```

Comment:
This confuses me. What are `time_controls` and a `time_controller`? Should this be `timestep_controls`? If so, what is the idea? A user can pass I assume a handle to a `timestep_controller` or parameters for one? I would just require a handle to a controller instead of this. Please add that the `timestep_controller` input is a handle to a controller.

Reply:
The paradigm I am aiming for here is to have the TimeStepController be in control of time in its entirety. Including termination times, whether (among solvers) time should synchronize, etc. 

(note: we could call it TimeController but that clashes with the succinct nature of things like AdaptiveTimeStepController)

The user then has two options, add the necessary options for a default TimeStepController, or pass a new TimeStepController, both accept time_controls as arguments, albeit, some controller might have more extensive control options than others.

Adjusted the documentation to:
```c++
params.AddOptionalParameter(
    "timestep_controller",
    0,
    "Timestep controller to use for timestepping. "
    "Cannot be supplied if the time_controls parameter is supplied");

  params.AddOptionalParameterBlock(
    "time_controls",
    chi::ParameterBlock{},
    "Parameters to pass to the default time_controller. If a "
    "timestep_controller is specified then this parameter can not be "
    "specified.");
  params.LinkParameterToBlock("time_controls",
                              "chi_physics::TimeStepController");
```

Reply:
I understand the idea, but I don't understand your "succinct nature of things" comment. I think if the `timestep_controller` controls time and doesn't just spit out a time step then it should be called `time_controller`. 

I still think that this is a weird input language and that the user should just be required to pass a `time_controller` handle to the solver, default or not. The user has to create the mesh outside of the solver, why not the time stepper too? It makes for a cleaner input in my opinion.